File: services/ingest-api/models/subscription.py
"""
Subscription Models.
"""

from enum import Enum
from pydantic import BaseModel
from datetime import datetime
from typing import Optional

class SubscriptionTier(str, Enum):
    FREE = "free"
    PRO = "pro"
    ENTERPRISE = "enterprise"

class Subscription(BaseModel):
    user_id: str
    tier: SubscriptionTier = SubscriptionTier.FREE
    current_period_start: datetime
    current_period_end: datetime
    event_count: int = 0
    limit: int

    @property
    def is_over_limit(self) -> bool:
        if self.tier == SubscriptionTier.ENTERPRISE:
            return False
        return self.event_count >= self.limit

TIER_LIMITS = {
    SubscriptionTier.FREE: 10_000,
    SubscriptionTier.PRO: 100_000,
    SubscriptionTier.ENTERPRISE: float('inf'),
}

File: web/src/api/subscription.ts
import { client } from './client';

export interface UsageStats {
  tier: 'free' | 'pro' | 'enterprise';
  usage: number;
  limit: number;
  percent_used: number;
}

export const getUsageStats = async (): Promise<UsageStats> => {
  const response = await client.get('/subscription/usage');
  return response.data;
};
